src/resnext.jl

```diff
@@ -0,0 +1,137 @@
+"""
+    resnextblock(inplanes, outplanes, cardinality, downsample = false)
+
+Create a basic residual block as defined in the paper for ResNeXt.
+([reference](https://arxiv.org/abs/1611.05431)).
+
+# Arguments:
+- `inplanes`: the number of input feature maps
+- `outplanes`: a list of the number of output feature maps for each convolution
+               within the residual block
+- `cardinality`: the number of groups to use for the convolution
+- `downsample`: set to `true` to downsample the input
+"""
+resnextblock(inplanes, outplanes, cardinality, downsample = false) = downsample ? 
+    Chain(
+        conv_bn((1, 1), inplanes, outplanes[1]; stride = 1, bias = false)...,
+        conv_bn((3, 3), outplanes[1], outplanes[2]; stride = 2, pad = 1, 
+                    groups = cardinality, bias = false)...,
+        conv_bn((1, 1), outplanes[2], outplanes[3], identity; stride = 1, bias = false)...) :
+    Chain(
+        conv_bn((1, 1), inplanes, outplanes[1]; stride = 1, bias = false)...,
+        conv_bn((3, 3), outplanes[1], outplanes[2]; stride = 1, pad = 1,
+                    groups = cardinality, bias = false)...,
+        conv_bn((1, 1), outplanes[2], outplanes[3], identity; stride = 1, bias = false)...)
```

Comment:
I believe the ResNeXt block is best understood as # of in channels, cardinality, and bottleneck width (see Fig. 3c in the paper). As such, the API for the block should probably be more like the following:
```julia
function resnextblock(inplanes, outplanes, cardinality, width, downsample = false)
    stride = downsample ? 2 : 1
    inner_channels = cardinality * width

    return Chain(conv_bn((1, 1), inplanes, inner_channels; stride = 1, bias = false)...,
                 conv_bn((3, 3), inner_channels, inner_channels;
                         stride = stride, pad = 1, bias = false, groups = cardinality)...,
                 conv_bn((1, 1), inner_channels, outplanes; stride = 1, bias = false)...)
end
```

Reply:
Yeah, I thought of doing away with the original ResNet style API but this is my first contribution so was a little unsure of how much I could change. The cardinality x width specification is much more characteristic for ResNeXt, I agree.